src/xtt.c

```diff
@@ -236,13 +225,32 @@ enftun_xtt_handshake(const char** server_hosts,
         goto finish;
     }
 
-    // 2) Make TCP connection to server.
-    ret = sock->ops.connect_any(sock, server_hosts, (char*) server_port);
-    if (ret < 0)
-    {
-        ret = 1;
-        goto finish;
-    }
+    // 2) Make network connection
+    // 2i) Try native sock type
+    enftun_tcp_native_init(&sock);
+    ret = sock.ops.connect_any(&sock, server_hosts, (char*) server_port, mark);
+
+    if (ret >= 0)
+        goto connect_success;
+    enftun_log_error("XTT Handshake: Native TCP sock connection failed\n");
```

Comment:
If the native connect failed, need to call `sock.ops.close(&sock);` before reusing `&sock` for the SCM attempt.  Otherwise, we'll leak the underlying native socket.

Reply:
Done